### DISCUSSION TITLE
Best practices for mock data creation.

### DISCUSSION THREAD

**Discussion by satodaiki** [pytest-dev/pytest#9130] [Ideas]
hi there.

I would like to talk about implementing the test code for the application I am writing.
Since the amount of code that creates mock data is large, the test code and the code that creates mock data are separated.
It looks like this:

```python
import pytest

# Array to store mock data
db = []

@pytest.fixture
def test_mock_data_1():
    # create mock data.
    db.append(1)

@pytest.mark.usefixture('test_mock_data_1')
def test_mock():
    assert db[0] == 1
```

In the above example (although quite simplified), fixture is used to create the mock data.
In fact, in the case I'm implementing, the mock data creation function is implemented in conftest.py and called from the test function like a normal fixture.

What I want to do is to further divide the mock data creation function defined in this fixture into files.
In other words, I want to define multiple conftest.py in one directory.

However, according to the specifications of pytest, conftest.py can only be placed once in one directory.
How should I put this fixture in multiple files?

**Comment by satodaiki** [comment]
One idea is to use `pytest_plugins` as follows.

data_1.py:
```python
import pytest

@pytest.fixture
def test_mock_data_1():
    # create mock data.
```

conftest.py:
```python
pytest_plugins = ('data_1', )
```

test_mock.py:
```python
import pytest

@pytest.mark.usefixture('test_mock_data_1')
def test_mock():
    assert db[0] == 1
```

However, in this case, pytest_plugins can be used only in conftest.py that exists in the root directory, so there is a disadvantage that the code of contest.py in the root becomes large in the case of multiple directory configurations as shown below.

```
test/
├─ sub/
│  ├─ conftest.py # Cannot be used "pytest_plugins" here
│  ├─ test_sub.py
│  ├─ __init__.py
├─ conftest.py # Only this file can apply "pytest_plugins".
├─ test_main.py 
├─ __init__.py

```

**Reply by jonathanmach** [reply]
In large projects with hundreds of fixtures, how to easily navigate to the file where which each fixture lives? Something link Ctrl+click to jump to the fixture definition...
I want our developer to be able to quickly investigate our fixtures and find others part of the same file
